src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordWithOcrToExistingCaseTest.java

```diff
@@ -0,0 +1,121 @@
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticatorFactory;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.logging.appinsights.SyntheticHeaders;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static java.time.Instant.now;
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.empty;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.definition.ExceptionRecordFields.ATTACH_TO_CASE_REFERENCE;
+
+@SpringBootTest
+@ActiveProfiles("nosb")
+class AttachExceptionRecordWithOcrToExistingCaseTest {
+
+    @Value("${test-url}")
+    String testUrl;
+
+    @Value("${document_management.url}")
+    String documentManagementUrl;
+
+    @Value("${document_management.context-path}")
+    String dmContextPath;
+
+    @Autowired CcdApi ccdApi;
+    @Autowired CcdCaseCreator ccdCaseCreator;
+    @Autowired CaseSearcher caseSearcher;
+    @Autowired EnvelopeMessager envelopeMessager;
+    @Autowired DocumentManagementUploadService dmUploadService;
+    @Autowired CcdAuthenticatorFactory ccdAuthenticatorFactory;
+
+    @Disabled
+    @Test
+    @SuppressWarnings("unchecked")
+    void should_update_case_with_ocr() throws Exception {
+        //given
+        CaseDetails existingCase = ccdCaseCreator.createCase(emptyList(), now());
+        String caseId = String.valueOf(existingCase.getId());
+
+        CaseDetails exceptionRecord = createExceptionRecord("envelopes/supplementary-evidence-with-ocr-envelope.json");
+        String ocrCountry = "sample_country"; // country from OCR data in exception record json loaded above
+
+        // when
+        sendAttachRequest(exceptionRecord, caseId);
+
+        // then
+        CaseDetails updatedCase = ccdApi.getCase(caseId, existingCase.getJurisdiction());
+
+        Map<String, String> address = (Map<String, String>) updatedCase.getData().get("address");
+        assertThat(address.get("country")).isEqualTo(ocrCountry);
+    }
+
+    private CaseDetails createExceptionRecord(String resourceName) throws Exception {
+        UUID poBox = UUID.randomUUID();
+
+        String dmUrl = dmUploadService.uploadToDmStore("doc.pdf", "documents/supplementary-evidence.pdf");
+
+        envelopeMessager.sendMessageFromFile(resourceName, "0000000000000000", null, poBox, dmUrl);
+
+        await("Exception record is created")
+            .atMost(30, TimeUnit.SECONDS)
+            .pollDelay(2, TimeUnit.SECONDS)
+            .until(() -> caseSearcher.findExceptionRecord(poBox.toString()).isPresent());
+
+        return caseSearcher.findExceptionRecord(poBox.toString()).get();
+    }
```

Comment:
We should start using `envelopeId` for searching created exception records during functional tests.

```java
String envelopeId = envelopeMessager.sendMessageFromFile(...);

await("Exception record is created")
    .atMost(30, TimeUnit.SECONDS)
    .pollInterval(2, TimeUnit.SECONDS)
    .until(() -> !findCasesByEnvelopeId(envelopeId).isEmpty());

List<CaseDetails> findCasesByEnvelopeId(String envelopeId) {
    return caseSearcher.search(
        SampleData.JURSIDICTION,
        SampleData.CONTAINER.toUpperCase() + "_" + CreateExceptionRecord.CASE_TYPE,
        ImmutableMap.of(
            "case.envelopeId", envelopeId
        )
    );
}
```

btw, we can improve await usage and return `CaseDetails` straight away:

```java
return away(...)
    .until(
        () -> findCasesByEnvelopeId(envelopeId),
        list -> !list.isEmpty()
    ).get(0)
```

Reply:
Searching by pobox is quite common as seen in https://github.com/hmcts/bulk-scan-orchestrator/pull/804/files
We can change it across the project on a separate PR.

Reply:
yeah would be good suggestion @doncem 
I don't mind doing it as a separate PR

Reply:
sounds like should've been done as part of #804 then 🤷‍♂ . since it's a new portion of code - can be done with envelope id initially. same thing was done for exception record during #744

Reply:
Well spotted, @doncem .